### PR TITLE
Make deploying SCORE possible even file/directory existent in a path …

### PR DESCRIPTION
**Pull request by inwonkim**
Make deploying SCORE possible even file/directory existent in a path where SCORE will be deployed.(#12)

### CODE REVIEW

File: iconservice/deploy/icon_score_deployer.py [hunk 1]
```diff
@@ -37,11 +39,9 @@ def deploy(self, address: Address, data: bytes, tx_hash: bytes):
         converted_tx_hash = f'0x{bytes.hex(tx_hash)}'
         install_path = os.path.join(score_root_path, converted_tx_hash)
         try:
-            if os.path.isfile(install_path):
-                raise ScoreInstallException(f'{install_path} is a file. Check your path.')
-            if os.path.isdir(install_path):
-                raise ScoreInstallException(f'{install_path} is a directory. Check {install_path}')
-            if not os.path.exists(install_path):
+            if os.path.exists(install_path):
```
Comment (leeheonseung) [inline iconservice/deploy/icon_score_deployer.py]: I think you ought to add mkdir logic previous that logic.
and you should add integrate-test about that

File: iconservice/deploy/icon_score_deployer.py [hunk 2]
```diff
@@ -37,11 +39,9 @@ def deploy(self, address: Address, data: bytes, tx_hash: bytes):
...
+                os.rename(install_path, f"{install_path}{int(time.time()*10**6)}_score_garbage")
```
Comment (leeheonseung) [inline iconservice/deploy/icon_score_deployer.py]: also that.

File: tests/integrate_test/test_existent_scores.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+from tests.integrate_test import root_clear
+from tests.integrate_test.test_integrate_base import TestIntegrateBase
+
+FILE_PATH = os.path.dirname(__file__)
+PROJECT_ROOT_PATH = os.path.join(FILE_PATH, '..', '..')
+BUILTIN_SCORE_SRC_ROOT_PATH = os.path.join(PROJECT_ROOT_PATH, 'iconservice', 'builtin_scores')
+GOVERNANCE_SCORE_PATH = os.path.join(BUILTIN_SCORE_SRC_ROOT_PATH, 'governance')
+
+
+class TestA(TestIntegrateBase):
```
Comment (leeheonseung) [inline tests/integrate_test/test_existent_scores.py]: you should rename, not like this

File: tests/test_icon_score_deployer.py [hunk 4]
```diff
@@ -63,11 +63,6 @@ def test_install(self):
         self.assertEqual(True, os.path.exists(install_path))
         self.assertTrue(installed_contents.sort() == file_path_list.sort())
 
-        # Case when the user install SCORE second time.
```
[the remark this answers is not in the document]
Comment (inwonkim) [reply tests/test_icon_score_deployer.py]: rolled back deploy_legacy method.
Added test for fail case

File: tests/integrate_test/test_integrate_existent_scores.py [hunk 5]
```diff
@@ -0,0 +1,379 @@
+# -*- coding: utf-8 -*-
```
Comment (leeheonseung) [inline tests/integrate_test/test_integrate_existent_scores.py]: It is regarded to deploy SCORE also.
so I want to add test that about audit enable circumstances

File: iconservice/deploy/icon_score_deploy_engine.py [hunk 6]
```diff
@@ -263,7 +266,15 @@ def _on_deploy(self,
             except FileExistsError:
                 pass
         else:
-            if context.get_revision() >= REVISION_2:
+            revision = context.get_revision()
+            if revision > REVISION_2:
+                score_root_path = os.path.join(self._icon_score_deployer.score_root_path,
```
Comment (Spl3en) [inline iconservice/deploy/icon_score_deploy_engine.py]: `os.path` is not necessary, `path` should be enough

File: tests/integrate_test/test_integrate_existent_scores.py (hunk 5 above)
Comment (inwonkim) [reply tests/integrate_test/test_integrate_existent_scores.py]: I added tests in audit enable circumstances.

[the remark this answers is not in the document]
Comment (inwonkim) [reply tests/integrate_test/test_integrate_existent_scores.py]: I added tests in audit enable circumstances.

File: iconservice/deploy/icon_score_deploy_engine.py [hunk 7]
```diff
@@ -12,22 +12,22 @@
...
 # limitations under the License.
-
 from os import path, symlink, makedirs
-
 from shutil import copytree
 from typing import TYPE_CHECKING, Callable
 
 from iconcommons import Logger
+
+from iconservice.deploy.icon_score_deploy_util import DirectoryNameConverter
```
Comment (leeheonseung) [inline iconservice/deploy/icon_score_deploy_engine.py]: Need to unify the import convention like the following from all modified files.

File: iconservice/deploy/icon_score_deploy_util.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
```
Comment (leeheonseung) [inline iconservice/deploy/icon_score_deploy_util.py]: I think you should combine this class and deployer class

File: iconservice/deploy/icon_score_deploy_util.py [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import time
+
+from iconservice import Address
```
Comment (leeheonseung) [inline iconservice/deploy/icon_score_deploy_util.py]: should fix from ..base.address import Address

File: iconservice/deploy/icon_score_deployer.py [hunk 10]
```diff
@@ -15,10 +15,11 @@
 # limitations under the License.
 import io
 import os
-import zipfile
 import shutil
+import zipfile
+
 from iconservice.base.address import Address
```
Comment (leeheonseung) [inline iconservice/deploy/icon_score_deployer.py]: Need to unify the import convention like the following from all modified files also.

File: tests/integrate_test/test_integrate_existent_scores.py [hunk 11]
```diff
@@ -0,0 +1,487 @@
...
+    def _setUp(self):
+        self.config.update_conf({ConfigKey.SERVICE: {ConfigKey.SERVICE_AUDIT: False,
+                                                     ConfigKey.SERVICE_FEE: False,
+                                                     ConfigKey.SERVICE_DEPLOYER_WHITELIST: False,
+                                                     ConfigKey.SERVICE_SCORE_PACKAGE_VALIDATOR: False}})
+        self.icon_service_engine = IconServiceEngine()
+        self.icon_service_engine.open(self.config)
+        self._genesis_invoke()
+
+    def _setUp_audit(self):
```
[the remark this answers is not in the document]
Comment (inwonkim) [reply tests/integrate_test/test_integrate_existent_scores.py]: I Divided tests to non_audit and audit.

File: tests/integrate_test/test_integrate_existent_scores.py [hunk 12]
```diff
@@ -0,0 +1,487 @@
...
+        self.assertNotEqual(original_governance_api, updated_governance_api)
+
+    # test when revision <= 2
+    def test_existent_score(self):
+        self._setUp()
+
+        # deploy SCORE
+        timestamp = int(time.time()*10**6)
+        tx1 = self._deploy_score('test_deploy_scores/install', 'sample_token', self._addr_array[0], ZERO_SCORE_ADDRESS,
+                                 timestamp, {"init_supply": hex(1000), "decimal": "0x12"})
```
[the remark this answers is not in the document]
Comment (inwonkim) [reply tests/integrate_test/test_integrate_existent_scores.py]: Extracted timestamp and deploy_params variables.

**Pull request merged**